### ISSUE TITLE
Fix weekday classical skeleton matching in datagen

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#5892]
See https://github.com/unicode-org/icu4x/pull/5826#issuecomment-2505702078

In the Finnish locale, we are resolving semantic skeletons to a pattern with `EEE` even though they should more likely be `ccc`. This is due to a potential limitation in the classical skeleton selection code.

**Comment by Manishearth** [comment]
Is this still an issue after neo?

**Comment by Manishearth** [comment]
Gemini thinks this is still an issue. I have not attempted to verify.

**Comment by sffc** [comment]
yes this is still an issue as far as I know. It is a data quality and conformance issue.

**Comment by sffc** [comment]
I had assumed that the ICU4X datetime pattern generator was dropping `c`, but it seems the issue was that the code assumed skeletons always used `E`, but some CLDR versions between ~41 and ~48.2 had data with a `c` in the skeleton (https://github.com/unicode-org/cldr/pull/5524).

**Comment by Manishearth** [comment]
Yeah that's also what I assumed and I started fixing the distance matching code (ICU4X does not understand that c and E are low distance) and stuff like that but it didn't have the right effect.
